### DISCUSSION TITLE
Can PaddleOCR detect multiple languages in a document?

### DISCUSSION THREAD

**Discussion by 99bits** [PaddlePaddle/PaddleOCR#13441] [Q&A]
Thank you for this great project and i am investigating PaddleOCR for a task that requires detection of multiple languages in the same document.   If this is possible, how to do this in the code?

Cheers

**Comment by GreatV** [comment]
PaddleOCR can't recognize multiple languages at the same time, it can only handle them separately.

**Comment by 99bits** [comment]
Ah thanks, so we have to do multiple passes.  Would  paddle OCR architecture be able to support multiple languages or would it take a huge overhaul of the engine?

**Reply by GreatV** [reply]
If you want to support more languages at the same time, you need to expand the character dictionary and collect a lot of data to retrain.

**Reply by 99bits** [reply]
Amazing, is there documentation that explains how to do this and the example dataset.  It also seems that currently English and Chinese is supported

**Reply by GreatV** [reply]
please refer to: https://github.com/PaddlePaddle/PaddleOCR/blob/main/doc/doc_en/training_en.md

**Reply by 99bits** [reply]
Thank you for the link.  Since you have a lot of practical experience with training PaddleOCR, can you share how much data was needed to train for Chinese and English.  And also the time it took for the training and the number of GPUs. Essentially I'm looking to see how much resources and time is needed to support multiple languages.  Also how should the dataset look like to support multiple detection of all languages that paddleOCR currently supports in the same document.

**Reply by UserWangZz** [reply]
Or can you implement a language classifier and use existing language recognition models for identification?
I think this might be another workaround.

**Reply by 99bits** [reply]
Have you tried this workaround before?

**Reply by UserWangZz** [reply]
sorry, I'm not.

**Reply by ashokparmar29** [reply]
> Or can you implement a language classifier and use existing language recognition models for identification? I think this might be another workaround.

Are the datasets used for training the recognition models for individual languages available as we can combine all datasets and label them, and use for training a language classifier.

**Reply by UserWangZz** [reply]
As I understand it, there are two ways to achieve your task:
1. Generate a multilingual dictionary for multiple languages, and then fine-tune the existing recognition model;
2. Classify the detected crop text area images by language, train a classification model, and use the existing language recognition model for recognition.

**Reply by ashokparmar29** [reply]
I mean, how can I get the datasets for training a language classifier? can you share links to download datasets used for training recognition models?

**Reply by snimavat** [reply]
the datalab models (Surya / Chandra) works with multiple language in same document, and supports good number of languages especially for indic languages, surya/and chandra are best models available right now.
